src/Components/OpenEnded/OpenEnded.js

```diff
@@ -2,71 +2,112 @@ import React, {useState} from 'react';
 import { useForm } from 'react-hook-form';
 import './OpenEnded.scss';
 
-function OpenEnded() { 
-
-    const [inputFieldOpenEnded, setInputFieldOpenEnded] = useState({
-        comp:'YesNoComponent',
-        qns:'Please enter question',
-        ansType:'short',
-        ans:''
+const OpenEnded=({ qno, data, id, isEdit = false, onUpdate})=> { 
+    
+    const onTitleUpdate = (value) => {
+        let tempData = { ...data };
+        tempData.title = value.target.value;
+        onUpdate(id,tempData);
+    }
+    const onButtonDetailsUpdate = (value) => {
+        let tempData = {...data};
+        tempData.ans= value.target.value;
+        onUpdate(id,tempData);
+    }
+    const selectTabs = (index, value) => {
+        let tempData = {...data};
+        tempData.ansType = value;
```

Comment:
see if you can reduce the above 3 method with one, as the purpose of all 3 r same and code is 90% same